tiledb/sm/consolidator/group_meta_consolidator.cc

```diff
@@ -81,48 +81,29 @@ Status GroupMetaConsolidator::consolidate(
       group_for_writes.open(QueryType::WRITE),
       throw_if_not_ok(group_for_reads.close()));
 
-  // Swap the in-memory metadata between the two groups.
+  // "Swap" the in-memory metadata between the two groups.
   // After that, the group for writes will store the (consolidated by
   // the way metadata loading works) metadata of the group for reads
-  Metadata* metadata_r;
-  auto st = group_for_reads.metadata(&metadata_r);
-  if (!st.ok()) {
-    throw_if_not_ok(group_for_reads.close());
-    throw_if_not_ok(group_for_writes.close());
-    return st;
-  }
-  Metadata* metadata_w;
-  st = group_for_writes.metadata(&metadata_w);
-  if (!st.ok()) {
-    throw_if_not_ok(group_for_reads.close());
-    throw_if_not_ok(group_for_writes.close());
-    return st;
-  }
-  metadata_r->swap(metadata_w);
-
-  // Metadata uris to delete
-  const auto to_vacuum = metadata_w->loaded_metadata_uris();
-
-  // Get the new URI name for consolidated metadata
-  URI new_uri = metadata_w->get_uri(group_uri);
-
-  // Close groups
-  RETURN_NOT_OK_ELSE(
-      group_for_reads.close(), throw_if_not_ok(group_for_writes.close()));
-  RETURN_NOT_OK(group_for_writes.close());
+  auto metadata_r = group_for_reads.metadata();
+  *(group_for_writes.metadata()) = *metadata_r;
+  URI new_uri = metadata_r->get_uri(group_uri);
+  const auto& to_vacuum = metadata_r->loaded_metadata_uris();
 
   // Write vacuum file
   URI vac_uri = URI(new_uri.to_string() + constants::vacuum_file_suffix);
-
   std::stringstream ss;
-  for (const auto& uri : to_vacuum)
+  for (const auto& uri : to_vacuum) {
     ss << uri.to_string() << "\n";
-
+  }
   auto data = ss.str();
   RETURN_NOT_OK(
       storage_manager_->vfs()->write(vac_uri, data.c_str(), data.size()));
   RETURN_NOT_OK(storage_manager_->vfs()->close_file(vac_uri));
 
+  // Close groups
+  RETURN_NOT_OK(group_for_reads.close());
+  RETURN_NOT_OK(group_for_writes.close());
+
```

Comment:
Same as array metadata consolidation, these closes need to be moved back to before the vacuum write.